engine/app/components/citizens_advice_components/input.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module CitizensAdviceComponents
+  class Input < Base
+    attr_reader :name, :label, :error_message, :hint, :width, :value
+
+    def initialize(name:, label:, type:, options: nil)
+      super
+      @name = name
+      @label = label
+      @type = type
+
+      set_options(options)
+    end
+
+    def set_options(options)
+      return unless options.present?
+
+      @error_message = options[:error_message]
+      @hint = options[:hint]
+      @optional = fetch_or_fallback_boolean(options[:optional], fallback: false)
+      @width = fetch_or_fallback(
+        allowed_values: allowed_width_values,
+        given_value: options[:width],
+        fallback: nil
+      )
+      @value = options[:value]
+      @additional_attributes = options[:additional_attributes]
+    end
+
+    def allowed_width_values
+      %w[2ch 4ch 8ch 16ch] << nil
```

Comment:
This syntax is fun, adding `nil` to the list? Wonder if this could be `[nil, :2ch, :4ch, :8ch, :16ch]`? I think using symbols here makes sense, consistent with the other components.

Reply:
Isn't it tho 😄   Agree it could be nicer.  Alas if you start a symbol with a number you have to enclose the symbol text in quotes, eg `:"2ch"` which feels a bit awkward to me and probably not immediately obvious to users?  Maybe it should just be a simple array `[nil, "2ch", "4ch", ...]`?

Reply:
OR could rename, eg `two-ch` and then just update the CSS.  Probably this I think.

Reply:
Ooh good point 👍  Yeah we should rename, the "2ch" format makes sense as implementation details but it doesn't need to be how we ask you to name them. Maybe go super explicit and say `:two_characters`? We can always map that to the terser class names behind the scenes, or rename.